source/common/router/header_parser.cc

```diff
@@ -64,16 +77,9 @@ ResponseHeaderParserPtr ResponseHeaderParser::parse(
   return response_header_parser;
 }
 
-// TODO(zuercher): If modifying this function to perform header substitutions, consider refactoring
-// this class and RequestHeaderParser to share overlapping code.
-void ResponseHeaderParser::evaluateResponseHeaders(Http::HeaderMap& headers) const {
-  for (const auto& header_value : headers_to_add_) {
-    if (header_value.append_) {
-      headers.addReferenceKey(header_value.header_, header_value.value_);
-    } else {
-      headers.setReferenceKey(header_value.header_, header_value.value_);
-    }
-  }
+void ResponseHeaderParser::evaluateResponseHeaders(
+    Http::HeaderMap& headers, const AccessLog::RequestInfo& request_info) const {
+  addHeaders(headers, request_info);
 
   for (const auto& header : headers_to_remove_) {
```

Comment:
Why can we remove response headers but not request headers?  Just a case of nobody (yet) needing to remove request headers?

Reply:
I think just that.

Reply:
Ooh. This made me think of an even better way to simplify all this.